robottelo/cli/base.py

```diff
@@ -374,6 +374,13 @@ def _construct_command(cls, options=None):
                 if val is True:
                     tail += u' --{0}'.format(key)
                 elif val is not False:
+                    if isinstance(val, list):
+                        val = ','.join(
+                            str(el.get('id') or el.get('name'))
```

Comment:
You can pass the list of ids or names in tests, like this:
```python
       result = Location.update({
           'id': loc['id'],
           'hostgroups': [hg['name'] for hg in new_host_groups],
       })
```
This line you've mentioned stands for different purpose. If you pass list of dictionaries (not names/ids), which basically means that you've passed entities, then we try to make a string from entities ids. If ids were not found (and there're some, which don't return their ids in cli) - try to make a string from enitity names. So it's just helper for a list of entities.